File: daily/Sliding Window/1208-Get-equal-substrings-within-budget.go
package daily

func equalSubstring(s string, t string, maxCost int) int {
	currentCost := 0
	left := 0
	n := len(s)
	maxLength := 0
	for right := 0; right < n; right++ {
		currentCost += abs(int(s[right]) - int(t[right]))
		for currentCost > maxCost {
			currentCost -= abs(int(s[left]) - int(t[left]))
			left++
		}
		maxLength = max(maxLength, right-left+1)
	}
	return maxLength
}

File: daily/Sliding Window/1208-Get-equal-substrings-within-budget.py
class Solution:
    def equalSubstring(self, s: str, t: str, maxCost: int) -> int:
        currentCost = 0
        left = 0
        n = len(s)
        maxLength = 0
        for right in range(n):
            currentCost += abs(ord(s[right]) - ord(t[right]))
            while currentCost > maxCost:
                currentCost -= abs(ord(s[left]) - ord(t[left]))
                left += 1
            maxLength = max(maxLength, right - left + 1)
        return maxLength